03_Modules/Configuration/src/module/api.ts

```diff
@@ -310,6 +325,125 @@ export class ConfigurationModuleApi
     return this._module.bootRepository.deleteByKey(request.query.stationId);
   }
 
+  @AsDataEndpoint(
+      Namespace.PasswordType,
+      HttpMethod.Post,
+      UpdateChargingStationPasswordQuerySchema,
+      UpdateChargingStationPasswordSchema,
+  )
+  async updatePassword(
+      request: FastifyRequest<{
+        Body: UpdateChargingStationPasswordRequest;
+        Querystring: UpdateChargingStationPasswordQueryString;
+      }>,
+  ): Promise<IMessageConfirmation> {
+    const stationId = request.body.stationId;
+    this._logger.debug(`Updating password for ${stationId} station`);
+    if (request.body.password && !isValidPassword(request.body.password)) {
+      return {success: false, payload: 'Invalid password'};
+    }
+    const password = request.body.password || generatePassword();
+
+    if (!request.body.setOnCharger) {
+      try {
+        await this.updatePasswordOnStation(password, stationId, request.query.callbackUrl);
+      } catch (error) {
+        this._logger.warn(`Failed updating password on ${stationId} station`, error);
+        return {success: false, payload: `Failed updating password on ${stationId} station`};
+      }
+    }
+    const variableAttributes = await this.updatePasswordForStation(password, stationId);
+    this._logger.debug(`Successfully updated password for ${stationId} station`);
+    return {success: true, payload: `Updated ${variableAttributes.length} attributes`};
+  }
+
+  private async updatePasswordOnStation(password: string, stationId: string, callbackUrl?: string): Promise<void> {
```

Comment:
Should I move this logic to a service layer instead? (I am uncertain about the convention; I have observed that many APIs are implemented directly within the API layer?)

Reply:
I think a service layer should be introduced and moved to there. API layer can keep logic on validating the request.